code/datums/quirks/negative_quirks/addict.dm

```diff
@@ -65,12 +61,35 @@
 		)
 	)
 
-/datum/quirk/item_quirk/junkie/remove()
+/datum/quirk/item_quirk/addict/junkie
+	name = "Junkie"
+	desc = "You can't get enough of hard drugs."
+	icon = FA_ICON_PILLS
+	value = -6
+	gain_text = span_danger("You suddenly feel the craving for drugs.")
+	medical_record_text = "Patient has a history of hard drugs."
+	hardcore_value = 4
+	quirk_flags = QUIRK_HUMAN_ONLY|QUIRK_PROCESSES
+	mail_goodies = list(/obj/effect/spawner/random/contraband/narcotics)
+	drug_flavour_text = "Better hope you don't run out..."
+
+/datum/quirk_constant_data/junkie
+	associated_typepath = /datum/quirk/item_quirk/addict/junkie
+	customization_options = list(/datum/preference/choiced/junkie)
+
+/datum/quirk/item_quirk/addict/junkie/add_unique(client/client_source)
+
+	var/addiction = client_source?.prefs.read_preference(/datum/preference/choiced/junkie)
+	if(addiction && (addiction != "Random"))
+		reagent_type = GLOB.possible_junkie_addictions[addiction]
+	return ..()
+
+/datum/quirk/item_quirk/addict/remove()
 	if(quirk_holder && reagent_instance)
 		for(var/addiction_type in subtypesof(/datum/addiction))
 			quirk_holder.mind.remove_addiction_points(addiction_type, MAX_ADDICTION_POINTS)
 
-/datum/quirk/item_quirk/junkie/process(seconds_per_tick)
+/datum/quirk/item_quirk/addict/process(seconds_per_tick)
```

Comment:
```suggestion
/datum/quirk/item_quirk/addict/junkie/process(seconds_per_tick)
```
Or if it's meant to be for the abstract quirk, move it above the declaration for `junkie`

Reply:
done